test-cypress/integration/cluster/cluster-configuration.spec.js

```diff
@@ -0,0 +1,277 @@
+import {ClusterPageSteps} from "../../steps/cluster/cluster-page-steps";
+import {GlobalOperationsStatusesStub} from "../../stubs/global-operations-statuses-stub";
+import {ClusterStubs} from "../../stubs/cluster/cluster-stubs";
+import {RemoteLocationStubs} from "../../stubs/cluster/remote-location-stubs";
+import {DeleteClusterDialogSteps} from "../../steps/cluster/delete-cluster-dialog-steps";
+import {ClusterConfigurationSteps} from "../../steps/cluster/cluster-configuration-steps";
+import {ModalDialogSteps} from "../../steps/modal-dialog-steps";
+import {ApplicationSteps} from "../../steps/application-steps";
+
+describe('Cluster configuration', () => {
+
+    let repositoryId;
+
+    beforeEach(() => {
+        repositoryId = 'cluster-repo' + Date.now();
+        GlobalOperationsStatusesStub.stubNoOperationsResponse(repositoryId);
+    });
+
+    it('Should display cluster configuration', () => {
+        // Given there is an existing cluster created
+        ClusterStubs.stubClusterConfig();
+        ClusterStubs.stubClusterGroupStatus();
+        ClusterStubs.stubClusterNodeStatus();
+        RemoteLocationStubs.stubRemoteLocationFilter();
+        RemoteLocationStubs.stubRemoteLocationStatusInCluster();
+        // Given I have opened the cluster management page
+        ClusterPageSteps.visit();
+        // When I click on edit properties
+        ClusterPageSteps.previewClusterConfig();
+        // Then I should see cluster configuration with 3 tabs
+        ClusterConfigurationSteps.getClusterConfig().should('be.visible');
+        ClusterConfigurationSteps.getTabs().should('have.length', 3);
+        ClusterConfigurationSteps.getActiveTab().should('have.text', 'Properties');
+        ClusterConfigurationSteps.getClusterPropertiesTabContent().should('be.visible');
+        ClusterConfigurationSteps.selectTab(1);
+        ClusterConfigurationSteps.getActiveTab().should('have.text', 'Nodes');
+        ClusterConfigurationSteps.getClusterNodesTabContent().should('be.visible');
+        ClusterConfigurationSteps.selectTab(2);
+        ClusterConfigurationSteps.getActiveTab().should('have.text', 'Multi-region');
+        ClusterConfigurationSteps.getMultiRegionTabContent().should('be.visible');
+    });
+
+    context('Properties', () => {
```

Comment:
Aplit all contexts in separate specs. In my experience, having these makes the spec quite unreadable and hard to maintain. Especially when each context has it's own tear-up different or complimentary to the main one. Not to mention that in time, the spec becomes huge

Reply:
done